apps/aecore/src/aec_block_micro_candidate.erl

```diff
@@ -247,10 +256,15 @@ add_txs_to_trees(MaxGas, Trees, [Tx | Txs], Acc, Env) ->
 report_failed_txs([])  -> ok;
 report_failed_txs(Txs) -> aec_tx_pool:failed_txs(Txs).
 
-used_gas(Block, Txs) ->
+used_gas(Block, Txs, Trees) ->
     Version = aec_blocks:version(Block),
     Height = aec_blocks:height(Block),
-    lists:foldl(fun(Tx, Acc) -> aetx:gas_limit(aetx_sign:tx(Tx), Height, Version) + Acc end, 0, Txs).
+    GasFun =
```

Comment:
Seems like this pattern appears in three places. Candidate for a separate function, perhaps?

Reply:
I think there are subtle differences though? But maybe that was in an earlier stage.

Reply:
Yes, perhaps